plugin.video.ecbtv/resources/lib/api.py

```diff
@@ -98,60 +109,107 @@ def _date_json(json_item):
+        thumbnail=None
+    )
+
+
+def counties():
+    for county in _soup('/county-championship/teams')('div', 'partners__item'):
+        team_id = int(os.path.basename(county.a['href']))
+        yield Entity(
+            name=county.a.text,
+            reference='cricket_team:{}'.format(team_id),
+            thumbnail=county.img['src']
+        )
+
+
+def player_categories():
+    for tab in _soup('/england/men/players').find_all(
+            'div', attrs={'data-ui-args': re.compile(r'{ "title": "\w+" }')}):
+        yield Entity(
+            name=tab['data-ui-tab'],
+            reference=None,
+            thumbnail=None
+        )
+
+
+def players(category='Test'):
+    soup = _soup('/england/men/players').find('div', attrs={'data-ui-tab': category})
+    for player in soup('section', 'profile-player-card'):
+        player_id = player.img['data-player']
+        yield Entity(
+            name=player.img['alt'],
+            reference='cricket_player:{}'.format(player_id),
+            thumbnail=PLAYER_THUMB_URL_FMT.format(category.lower(), player_id)
         )
 
 
-def search_results(term, start=0, size=10):
+def _video(video):
+    return Video(
+        title=video['title'],
+        url=HLS_URL_FMT.format(video['mediaId']),
+        thumbnail=_thumbnail_variant(video),
+        date=_date_json(video),
+        duration=video['duration']
+    )
+
+
+def _videos(videos_json):
+    '''Generator for all videos from a particular page'''
+    for video in videos_json['content']:
+        yield _video(video)
+
+
+def videos(reference=None, page=1, page_size=10):
+    videos_json = requests.get(_video_list_url(reference, page, page_size)).json()
+    npages = videos_json['pageInfo']['numPages']
+    return _videos(videos_json), npages
+
+
+def _search_results(search_results_json):
     '''Generator for videos matching a search term'''
-    results = requests.get(_search_url(term, start, size)).json()['hits']['hit']
+    results = search_results_json['hits']['hit']
     for result in results:
         video = result['response']
-        yield Video(
-            title=video['title'],
-            url=HLS_URL_FMT.format(video['mediaId']),
-            thumbnail=video['imageUrl'],
-            date=_date_json(video),
-            duration=video['duration']
-        )
+        yield _video(video)
+
+
+def search_results(term, page=1, page_size=10):
+    search_results_json = requests.get(_search_url(term, page, page_size)).json()
+    total = search_results_json['hits']['found']
+    npages = int(math.ceil(float(total) / page_size))
+    return _search_results(search_results_json), npages
 
 
-def _print_all_videos():
+def _print_team_videos():
     '''Test function to print all categories and videos'''
-    for title, path in categories():
-        print '{} ({})'.format(title, path)
-        for video in videos(path):
+    for team in [england()] + list(counties()):
+        print '{} ({})'.format(team.name, team.reference)
```

Comment:
Please move all prints to xbmc.log with a min. log level of LOGDEBUG, THX
https://codedocs.xyz/xbmc/xbmc/group__python__xbmc.html#ga3cd3eff2195727144ad6fbabd4744961